supervision/annotators/core.py

```diff
@@ -1669,27 +1674,80 @@ def annotate(
         ![triangle-annotator-example](https://media.roboflow.com/
         supervision-annotator-examples/triangle-annotator-example.png)
         """
-        xy = detections.get_anchors_coordinates(anchor=self.position)
-        for detection_idx in range(len(detections)):
-            color = resolve_color(
-                color=self.color,
-                detections=detections,
-                detection_idx=detection_idx,
-                color_lookup=self.color_lookup
-                if custom_color_lookup is None
-                else custom_color_lookup,
-            )
-            tip_x, tip_y = int(xy[detection_idx, 0]), int(xy[detection_idx, 1])
-            vertices = np.array(
-                [
-                    [tip_x - self.base // 2, tip_y - self.height],
-                    [tip_x + self.base // 2, tip_y - self.height],
-                    [tip_x, tip_y],
-                ],
-                np.int32,
-            )
+        if self.outline_thickness:
```

Comment:
This is a lot of logic for a small change.

A more efficient method involve using `cv2.polylines`.
There you can specify thickness and draw another shape behind the original triangle.

I wouldn't expect to have two very different if-else branches.

Reply:
Actually the original triangles are little small :)
No problem, I will add cv2.polygons and the thickness parameter.